components/scream/src/share/util/scream_column_ops.hpp

```diff
@@ -79,218 +87,102 @@ class ColumnOps {
-        pack_type tmp = ekat::shift_left(x_i(k+1)[0], x_i(k));
-        tmp += x_i(k);
-        tmp /= 2.0;
-        combine<CM>(tmp, x_m(k), alpha, beta);
-      };
-      if (NUM_MID_PACKS==NUM_INT_PACKS) {
-        const auto LAST_PACK =  NUM_MID_PACKS-1;
-
-        // Use SIMD operations only on NUM_MID_PACKS-1, since mid pack
-        // does not have a 'next' one
-        team_parallel_for(team,NUM_MID_PACKS-1,shift_and_avg);
-
-        team_single (team, [&]() {
-          // Last level pack treated separately, since int pack k+1 does not exist.
-          // Shift's first arg is the scalar to put in the "empty" spot at the end.
-          // In this case, we don't need it, so just uze zero.
-          const auto& xi_last = x_i(LAST_PACK);
-          pack_type tmp = ekat::shift_left(zero(), xi_last);
-
-          tmp += xi_last;
-          tmp /= 2.0;
-          combine<CM>(tmp, x_m(LAST_PACK), alpha, beta);
-        });
-      } else {
-        // We can use SIMD operations on all NUM_MID_PACKS mid packs,
-        // since x_i(k+1) is *always* fine
-        team_parallel_for(team,NUM_MID_PACKS,shift_and_avg);
-      }
-    }
+    compute_midpoint_values_impl<CM,InputProvider,ScalarT>(team,num_mid_levels,x_i,x_m,alpha,beta);
   }
 
   // Compute X at level interfaces, given X at level midpoints and top or bot bc.
   // Note: with proper bc, then x_int(x_mid(x_int))==x_int.
-  template<bool FixTop, typename InputProvider = DefaultProvider>
+  template<bool FixTop, typename InputProvider, typename ScalarT>
   KOKKOS_INLINE_FUNCTION
   static void
   compute_interface_values (const TeamMember& team,
                             const int num_mid_levels,
                             const InputProvider& x_m,
                             const scalar_type& bc,
-                            const view_1d<pack_type>& x_i)
+                            const view_1d<ScalarT>& x_i)
   {
-    compute_interface_values_impl<pack_size,FixTop>(team,num_mid_levels,x_m,bc,x_i);
+    // Sanity checks
+    debug_checks<InputProvider>(num_mid_levels+1,x_i);
+
+    compute_interface_values_impl<FixTop>(team,num_mid_levels,x_m,bc,x_i);
   }
 
   // Given X at level interfaces, compute dX at level midpoints.
-  template<CombineMode CM = CombineMode::Replace,
-           typename InputProvider = DefaultProvider>
+  template<typename InputProvider, typename ScalarT>
   KOKKOS_INLINE_FUNCTION
   static void
   compute_midpoint_delta (const TeamMember& team,
                           const int num_mid_levels,
                           const InputProvider& x_i,
-                          const view_1d<pack_type>& dx_m,
+                          const view_1d<ScalarT>& dx_m,
+                          const scalar_type alpha = one(),
+                          const scalar_type beta = zero())
+  {
+    compute_midpoint_delta<CombineMode::Replace>(team,num_mid_levels,x_i,dx_m,alpha,beta);
+  }
+
+  // Given X at level interfaces, compute dX at level midpoints.
```

Comment:
Maybe describe CombineMode and its supported options (unless they aren't a fixed set).